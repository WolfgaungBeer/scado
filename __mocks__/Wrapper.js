import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Theme, defaultTheme } from '../scado';

const mockStore = configureStore();
const store = mockStore({
    scado: {
        theme: defaultTheme,
        toasts: [
            { id: '1', type: 'success', content: 'Text', onClose: () => {} },
            { id: '2', type: 'error', content: 'Text', onClose: () => {} },
        ],
    },
});


const Wrapper = ({ children }) => ( // eslint-disable-line
    <Provider store={store}>
        <Theme>{children}</Theme>
    </Provider>
);

export default Wrapper;
