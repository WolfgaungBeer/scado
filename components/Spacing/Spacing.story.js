import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import Spacing from './';


storiesOf('Spacing', module)
    .add('default', () => (
        <Spacing />
    ))
    .add('with props', () => (
        <Spacing scale={text('scale', 'l')} />
    ));
