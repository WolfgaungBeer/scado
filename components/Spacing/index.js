import styled from 'styled-components';
import { string } from 'prop-types';

const propTypes = {
    scale: string,
};

const defaultProps = {
    scale: undefined,
};

const Spacing = styled.div`
    margin-top: ${props => props.theme.scale[props.scale] || props.theme.scale.m};
`;

Spacing.propTypes = propTypes;
Spacing.defaultProps = defaultProps;

export default Spacing;
