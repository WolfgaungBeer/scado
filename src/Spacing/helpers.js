export const getStyle = () => (props) => {
    const { scale, left, bottom, right } = props;
    let pos = 'top';
    if (left) pos = 'left';
    if (bottom) pos = 'bottom';
    if (right) pos = 'right';
    return `padding-${pos}: ${props.theme.scale[scale] || props.theme.scale.m};`;
};
