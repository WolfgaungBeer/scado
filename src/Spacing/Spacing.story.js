import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import Spacing from './';


storiesOf('Spacing', module)
    .add('default', () => (
        <Spacing>Spacing</Spacing>
    ))
    .add('with props', () => (
        <Spacing
            scale={text('scale', 'l')}
            top={boolean('top', false)}
            left={boolean('left', false)}
            bottom={boolean('bottom', false)}
            right={boolean('right', false)}
        >
            Spacing
        </Spacing>
    ));
