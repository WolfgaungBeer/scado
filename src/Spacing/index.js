import styled from 'styled-components';
import { string, bool, node } from 'prop-types';
import { getStyle } from './helpers';

const propTypes = {
    scale: string,
    top: bool,
    left: bool,
    bottom: bool,
    right: bool,
    children: node,
};

const defaultProps = {
    scale: undefined,
    top: undefined,
    left: undefined,
    bottom: undefined,
    right: undefined,
    children: undefined,
};

const Spacing = styled.div`
    ${getStyle()}
`;

Spacing.propTypes = propTypes;
Spacing.defaultProps = defaultProps;

export default Spacing;
