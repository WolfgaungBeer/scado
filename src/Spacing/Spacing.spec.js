import React from 'react';
import renderer from 'react-test-renderer';
import Wrapper from '../../__mocks__/Wrapper';
import Spacing from './';

it('renders the Spacing Component correctly', () => {
    const tree1 = renderer.create(
        <Wrapper>
            <Spacing>Spacing</Spacing>
        </Wrapper>,
    ).toJSON();
    expect(tree1).toMatchSnapshot();

    const tree2 = renderer.create(
        <Wrapper>
            <Spacing scale="xxl" top>Spacing</Spacing>
        </Wrapper>,
    ).toJSON();
    expect(tree2).toMatchSnapshot();

    const tree3 = renderer.create(
        <Wrapper>
            <Spacing scale="xxl" left>Spacing</Spacing>
        </Wrapper>,
    ).toJSON();
    expect(tree3).toMatchSnapshot();

    const tree4 = renderer.create(
        <Wrapper>
            <Spacing scale="xxl" bottom>Spacing</Spacing>
        </Wrapper>,
    ).toJSON();
    expect(tree4).toMatchSnapshot();

    const tree5 = renderer.create(
        <Wrapper>
            <Spacing scale="xxl" right>Spacing</Spacing>
        </Wrapper>,
    ).toJSON();
    expect(tree5).toMatchSnapshot();
});
