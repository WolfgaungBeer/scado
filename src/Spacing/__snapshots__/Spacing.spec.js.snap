// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Spacing Component correctly 1`] = `
<div
  className="sc-cSHVUG koibno"
  scale={undefined}
>
  Spacing
</div>
`;

exports[`renders the Spacing Component correctly 2`] = `
<div
  className="sc-cSHVUG dFHALJ"
  scale="xxl"
>
  Spacing
</div>
`;

exports[`renders the Spacing Component correctly 3`] = `
<div
  className="sc-cSHVUG hzwtXh"
  scale="xxl"
>
  Spacing
</div>
`;

exports[`renders the Spacing Component correctly 4`] = `
<div
  className="sc-cSHVUG kAusrL"
  scale="xxl"
>
  Spacing
</div>
`;

exports[`renders the Spacing Component correctly 5`] = `
<div
  className="sc-cSHVUG fOKbxR"
  scale="xxl"
>
  Spacing
</div>
`;
